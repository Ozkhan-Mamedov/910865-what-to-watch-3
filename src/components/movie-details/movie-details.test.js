import React from "react";
import renderer from "react-test-renderer";

import MovieDetails from "./movie-details";

const film = {
  name: `Film#1`,
  picture: `picture#1`,
  genre: `genre#1`,
  id: 1,
  releaseDate: `2020`,
  ratingScore: 1,
  ratingsNumber: 2,
  director: `Director#1`,
  starring: [`Actor#1`, `Actor#2`, `Actor#3`, `Actor#4`],
  description: [`Test paragraph #1.`, `Test paragraph #2.`],
};

it(`MovieDetails component renders correctly`, () => {
  const tree = renderer
    .create(<MovieDetails film={film} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
