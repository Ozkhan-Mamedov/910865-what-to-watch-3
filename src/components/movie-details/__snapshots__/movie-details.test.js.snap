// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "background color#1",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Film#1"
          src="background image#1"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head movie-card__head"
        style={Object {}}
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="main.html"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block"
          >
            <a
              className="user-block__link"
              href="sign-in.html"
              onClick={[Function]}
            >
              Sign in
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Film#1
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              genre#1
            </span>
            <span
              className="movie-card__year"
            >
              2020
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="preview image#1"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
                onClick={[Function]}
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-card__desc"
          >
            <div
              className="movie-rating"
            >
              <div
                className="movie-rating__score"
              >
                2
              </div>
              <p
                className="movie-rating__meta"
              >
                <span
                  className="movie-rating__level"
                >
                  Bad
                </span>
                <span
                  className="movie-rating__count"
                >
                  1
                   ratings
                </span>
              </p>
            </div>
            <div
              className="movie-card__text"
            >
              <p>
                Test paragraph #1.
              </p>
              <p
                className="movie-card__director"
              >
                <strong>
                  Director: 
                  Director#1
                </strong>
              </p>
              <p
                className="movie-card__starring"
              >
                <strong>
                  Starring: 
                  Actor#1, Actor#2, Actor#3, Actor#4
                   and other
                </strong>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Film#1"
              height="175"
              src="picture#1"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Film#1
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
